spec/requests/api/v2/providers/courses/update_gsce_requirements_spec.rb

```diff
@@ -0,0 +1,137 @@
+  let(:site_status)       { build(:site_status, :findable, site: site) }
+  let(:site)              { build(:site) }
+  let(:primary_subject)   { build(:subject, :primary) }
+  let(:updated_course)    {
+    build :course,
+          course_code: course.course_code,
+            provider: provider,
+            site_statuses: [site_status],
+            subjects: [primary_subject],
+            **gcse_requirements
+  }
+
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  let(:permitted_params) do
+    %i[english maths science]
+  end
+
+  context "course has updated gcse requirements" do
+    let(:gcse_requirements) do
+      {
+        english: 'expect_to_achieve_before_training_begins',
+        maths: 'expect_to_achieve_before_training_begins',
+        science: 'expect_to_achieve_before_training_begins'
+      }
+    end
+
+    before do
+      updated_course.id = course.id
+      perform_request(updated_course)
+    end
+
+    it "returns http success" do
+      expect(response).to have_http_status(:success)
+    end
+
+    it "updates the english attribute to the correct value" do
+      expect(course.reload.english).to eq(updated_course.english)
+    end
+
+    it "updates the maths attribute to the correct value" do
+      expect(course.reload.maths).to eq(updated_course.maths)
+    end
+
+    it "updates the science attribute to the correct value" do
+      expect(course.reload.science).to eq(updated_course.science)
+    end
+  end
+
+  context "course has no updated gcse requirements" do
+    context "with values passed into the params" do
+      let(:gcse_requirements) do
+        {
+          english: 'must_have_qualification_at_application_time',
+          maths: 'must_have_qualification_at_application_time',
+          science: 'must_have_qualification_at_application_time'
+        }
+      end
+
+      before do
+        updated_course.id = course.id
+        perform_request(updated_course)
+      end
+
+      it "returns http success" do
+        expect(response).to have_http_status(:success)
+      end
+
+      it "does not change english attribute" do
+        expect(course.reload.english).to eq(course.english)
+      end
+
+      it "does not change maths attribute" do
+        expect(course.reload.maths).to eq(course.maths)
+      end
+
+      it "does not change the science attribute" do
+        expect(course.reload.science).to eq(course.science)
+      end
+    end
+
+    context "with no values passed into the params" do
+      let(:gcse_requirements) { {} }
+
+      before do
+        updated_course.id = course.id
+        perform_request(updated_course)
+      end
+
+      it "returns http success" do
+        expect(response).to have_http_status(:success)
+      end
+
+      it "does not change english attribute" do
+        expect(course.reload.english).to eq(course.english)
```

Comment:
isn't that always true? I think you need to keep a copy of what it was somewhere

Reply:
You are correct. Have addressed.

Reply:
Thanks for the feedback!